dlfcn.h

```diff
@@ -55,7 +57,10 @@ extern "C" {
 /* Specifies the next object after this one that defines name. */
 #define RTLD_NEXT       ((void *)-1)
 
-/* Open a symbol table handle. */
+/* Open a symbol table handle (file path is in wide character). */
+DLFCN_EXPORT void *wdlopen(const wchar_t *file, int mode);
+
+/* Open a symbol table handle (ASCII file path only). */
 DLFCN_EXPORT void *dlopen(const char *file, int mode);

```

Comment:
This is incorrect. File name is not in ASCII as it can contain also other non-ASCII characters which are encoded according to current code page.

Reply:
So, is it possible to use `dlopen` with Unicode file inputs as well?

Reply:
No, it is not.

Reply:
Seems that you are mixing more things together. To make it clear:

ASCII = 7bit encoding, match exactly first 128 UNICODE code points
Latin1 = ISO-8859-1 = fixed 8bit encoding, match exactly first 256 UNICODE code points, therefore extension of ASCII
Windows-1252 = fixed 8bit encoding, extension of ASCII, but not same as Latin1
ANSI encoding (in windows terminology) = current codepage encoding in Windows system, user can change it, Windows-1252 is one example
UNICODE (in windows terminology) = UTF-16
UNICODE code point = number between 0 and 0x10FFFF (except numbers which matches UTF-16 surrogate pairs)
UTF-16 = way how to encode UNICODE code points into uint16_t[] array; variable width
UTF-8 = way how to encode UNICODE code points into uint8_t[] array; extension of ASCII; variable width
UTF-32 = UCS-4 = way how to encode UNICODE code points into uint32_t[] array; fixed width
UCS-2 = way how to encode UNICODE code points smaller then 0x10000 into uint16_t[] array; fixed width

So, dlopen cannot handle all UNICODE file names, but also is not ASCII-only, as it can handle UNICODE code points which are present in current code page.

Reply:
WIN32 APIs tend to come in `BlahA()` and `BlahW()` forms, with the `W` forms using `wchar_t` strings containing UTF-16 (historically UCS-2), and the `A` forms using `char` with whatever encoding the current codepage would use with code unit size of 1 byte (hopefully everyone can switch to using only UTF-8 someday).  I assume the `A` is meant to evoke ASCII, though it isn't ASCII.  So the comment is wrong, for sure, but I think that mistake is easy to make :)

Reply:
> `A` forms using `char` with whatever encoding the current codepage would use with code unit size of 1 byte

This is not truth. Some Windows -A encodings are also variable one to two bytes. Moreover new builds of Windows 10 has beta option which switch -A forms into UTF-8 therefore they are up to the 4 bytes.

>  I assume the `A` is meant to evoke ASCII

Nope, -A means ANSI, not ASCII. ASCII is 7bit, ANSI (in windows term) is fixed 8bit or variable 8+ bits.

Reply:
"code unit" just refers to the smallest addressable word size in a codeset/encoding.  For UTF-8 the code unit size is 1 byte, but a codepoint can take up more than 1 code unit because UTF-8 is a multi-byte encoding. For UTF-16 the code unit size is 2 bytes, though again, a codepoint can require more than 1 code unit to encode.  For UTF-32 the code unit size is 4 and you never need more than 1 code unit to encode a codepoint in UTF-32.

Reply:
You're right about the `A` -- I'd forgotten that detail.